qiskit/primitives/utils.py

```diff
@@ -113,6 +113,16 @@ def final_measurement_mapping(circuit: QuantumCircuit) -> dict[int, int]:
     return mapping
 
 
+def _circuit_key(circuit: QuantumCircuit) -> tuple:
```

Comment:
Could you write a short description of this function though it's private?  We may need it for maintenance.
It might be also good to write that we can remove it when `QuantumCircuit.__hash__` is introduced.